R/slice.R

```diff
@@ -462,7 +467,30 @@ sample_int <- function(n, size, replace = FALSE, wt = NULL, call = caller_env())
   }
 }
 
-smaller_ranks <- function(x, y) {
-  sum(min_rank(x) <= y, na.rm = TRUE)
+slice_rank_idx <- function(
+    order_by,
+    size,
+    with_ties = TRUE,
+    direction = c("asc", "desc"),
+    na_rm = FALSE,
+    call = caller_env()
+) {
+  direction <- arg_match(direction, error_call = call)
+  # puts missing values at the end
+  na_value <- if (direction == "asc") "largest" else "smallest"
+  ties <- if (with_ties) "min" else "sequential"
+
+  ranks <- vctrs:::vec_rank(order_by,
+    ties = ties,
+    direction = direction,
+    na_value = na_value
+  )
+
+  keep <- ranks <= size
+  if (isTRUE(na_rm)) {
```

Comment:
```suggestion
  if (na_rm) {
```

I think another PR ensured that the generics assert this is a boolean, so we can avoid this per group check